index.ts

```diff
@@ -20,64 +21,83 @@ app.use(express.json());
 
 app.post("/webhook", async (req, res) => {
   console.log("req receieved");
-  const signature = req.headers["x-hub-signature"];
+  const signature = req.headers["x-hub-signature"] as string;
+  console.log(signature)
+  console.log("header",req.headers)
   const payload = JSON.stringify(req.body);
 
-  const hmac = crypto.createHmac("sha1", process.env.GITHUB_SECRET);
+  const hmac = crypto.createHmac("sha1", secret);
   const calculatedSignature = `sha1=${hmac.update(payload).digest("hex")}`;
+  console.log("cals", calculatedSignature)
 
   if (crypto.timingSafeEqual(Buffer.from(signature), Buffer.from(calculatedSignature))) {
-    const { result, respMessage } = await getBranchStatus();
+    const {result, respMessage } = await getBranchStatus(req.body);
     console.log("Result: ", result);
-    res.status(result).send(respMessage);
+    res.status(200).send({ result, respMessage });
   } else {
     res.status(400).send("Invalid Github signature");
   }
 });
 
-app.listen(process.env.PORT, () => {
+interface BranchStatus {
+  result:  number | string;
+  respMessage: string ;
+}
+
+app.listen(port, () => {
   console.log(`Server listening on port ${port}`);
 });
 
-const executeCmd = async (cmd) => {
+const executeCmd = async (cmd: string) => {
   try {
-    const { stdout, stderr } = await exec(cmd);
+    const {stdout, stderr} = await exec(cmd);
     return stderr + "\n" + stdout;
-  } catch (error) {
+  } catch (error: any) {
     console.error(`exec error: ${error}`);
-    throw new Error(stderr + "\n" + stdout);
+    throw new Error(error.stderr + "\n" + error.stdout);
   }
 };
 
-const getBranchStatus = async (req) => {
+async function getBranchStatus(req: any): Promise<BranchStatus> {
   console.log("Webhook received successfully");
 
   const branchName = req.body?.ref?.split("/").pop();
+
   if (!branchName) {
-    return 400, "Branch name not found in the request.";
+    return { result: 400, respMessage: "Branch name not found in the request." };
+  }
+
+  if (branchName === process.env.BRANCH_NAME) {
+    const { status, message } = await buildProject();
+    return { result: status, respMessage: message };
+  } else {
+    return { result: 200, respMessage: "Build not required." };
   }
-  return branchName === process.env.BRANCH_NAME ? await buildProject() : 202, "Build not required.";
 };
 
 const isUpdateRequired = () => {
   const currentTime = Date.now();
-  isMindmapUpdated = (currentTime - mindmapBuildTime) / 1000 / 60 > process.env.MINDMAP_UPDATE_TIME_INTERVAL ? true : false;
-  isDocumentationWebsiteUpdated = (currentTime - documentationWebsiteBuildTime) / 1000 / 60 > process.env.DOCUMENTATION_WEBSITE_UPDATE_TIME_INTERVAL ? true : false;
+  const mindMapUpdateInterval = parseInt("process.env.MINDMAP_UPDATE_TIME_INTERVAL", 10); // converted to number uusing variable 
+  const documentationWebsiteUpdateInterval = parseInt("process.env.DOCUMENTATION_WEBSITE_UPDATE_TIME_INTERVAL", 10); // converted to num using variable 
```

Comment:
this makes no sense, why are you converting static strings to numbers?

Reply:
Hi, 
When attempting to compare I am returned an error "Operator '>' cannot be applied to types 'number' and 'string'". I attempted to correct the error by converting the variable to a number. It is my understanding that numbers and strings cannot be compared without type conversion.  After doing more research it seems as type conversion could be needed to compare both the static string and numbers. Do you happen to have an alternate method to compare the static string with numbers?

Reply:
If you were converting a variable I wouldn't have complained ;) You are comparing a static string instead

Reply:
Would I need to apply a method to the static string for the comparison? I found a method called localeString().

Reply:
I attempted to convert the given value to a string,
is this more along the lines? 
const mindMapUpdateInterval = "process.env.MINDMAP_UPDATE_TIME_INTERVAL";
 isMindmapUpdated = ((currentTime - mindmapBuildTime) / 1000 / 60).toString() > mindMapUpdateInterval;

Reply:
corrected variable